can_validator/MY17_Can_Library_c.py

```diff
@@ -0,0 +1,168 @@
+"""
+Generate MY17_Can_Libary.c file.
+Run this file to write just MY17_Can_Libary.c or main.py to write all files.
+"""
+import sys
+sys.path.append("ParseCAN")
+import ParseCAN
+from math import ceil
+from common import can_lib_c_path, spec_path, can_lib_c_base_path, can_lib_c_special_cases_path, unused_messages
+
+
+def write(output_path, spec_path, base_path, special_cases_path, unused_messages):
+    """
+    Generate MY17_Can_Libary.c file.
+
+    :param output_path: file to be written to
+    :param spec_path: CAN spec path
+    :param base_path: File with template code that's not autogenerated
+    :param special_cases_path: File with code for special cases of functions that aren't based off of CAN spec
+    :param unused_messages: list of messages that are in the CAN spec but unused
+    """
+    spec = ParseCAN.spec.can(spec_path)
+    with open(output_path, 'w') as f:
+        # Copy over base
+        with open(base_path) as base:
+            lines = base.readlines()
+            f.writelines(lines)
+
+        # Write switch statement
+        f.write(
+            "Can_MsgID_T Can_MsgType(void) {\n" +
```

Comment:
I think it would be better to get familiar with multi-line strings and use these for MY18. Now this works, but consider that the docstring format of
```
"""
<body>
"""
```
also preserves newlines and indents.